File: solve/0813_명예의 전당 (1)/postforty/명예의 전당 (1).py
# https://school.programmers.co.kr/learn/courses/30/lessons/138477
def solution(k, score):
    answer = [score[0]]
    score_board = [score[0]]
    for i in score[1:]:
        score_board.append(i)
        score_board.sort()
        if len(score_board) > k:
            score_board = score_board[1:]
        answer.append(score_board[0])

    return answer


print(solution(3, [10, 100, 20, 150, 1, 100, 200]))  # [10, 10, 10, 20, 20, 100, 100]

File: solve/0813_명예의 전당 (1)/postforty/명예의 전당 (1).js
// https://school.programmers.co.kr/learn/courses/30/lessons/138477
function solution(k, score) {
  var answer = [score[0]];
  let scoreBoard = [score[0]];
  score = [...score.slice(1, score.length)];
  score.forEach((v) => {
    scoreBoard.push(v);
    scoreBoard.sort((a, b) => a - b);
    if (scoreBoard.length > k) {
      scoreBoard = scoreBoard.slice(1, scoreBoard.length);
    }
    answer.push(scoreBoard[0]);
  });

  return answer;
}

console.log(solution(3, [10, 100, 20, 150, 1, 100, 200])); // [10, 10, 10, 20, 20, 100, 100]
